File: backend/Ecommerce/src/main/java/com/Ojt/Ecommerce/entity/Category.java
package com.Ojt.Ecommerce.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "category")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private List<ProductHasCategory> productCategories;

    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<BrandHasCategory> brandCategories;

    @ManyToOne
    @JoinColumn(name = "parent_id")
    private Category parent;

    @OneToMany(mappedBy = "parent", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Category> children;

    @Column(name = "image", columnDefinition = "TEXT")
    private String image;

    @Column(name = "icon_url", columnDefinition = "TEXT")
    @JsonProperty("iconUrl")
    private String iconUrl;

    @Column(name = "icon_class", length = 100)
    @JsonProperty("iconClass")
    private String iconClass;

    @Column(name = "status", columnDefinition = "INT DEFAULT 1")
    private Integer status;

    @PrePersist
    public void prePersist() {
        if (status == null) {
            status = 1;
        }
    }
}

File: frontend/Ecommerce/src/app/category.ts
export interface Category {
    id: number;
    name: string;
    image?: string;
    parentId?: number;
    status?: number;
    iconUrl?: string;
    iconClass?: string;
}

export interface CategoryDTO{
    cateNames : String[];
    brandId : number;
    brandName : String;
    parentId?: number;
    image?: String;
    icon?: string;
    iconUrl?: string;
    iconClass?: string;
}

export interface CategoryListDTO {
    id: number;
    name: string;
    image: string;
    subcategories: CategoryListDTO[];
  }
  
export interface CategoryTreeDTO {
    id: number;
    name: string;
    image?: string;
    status: number;
    children?: CategoryTreeDTO[];
}
  
export interface SubCategoryDTO {
    parentId: number;
    subCategoryNames: string[];
  }  